### PR TITLE
Remove StorageRequestFailedException

### LINKED ISSUE #8144: [P0] Storage: Make StorageRequestFailedException internal
~~Note: blocked on the Azure.Core decision tracked by #8143~~ (#8152)

https://apiview.azurewebsites.net/Assemblies/Review/b48340c11dab4e88b3b2bd269bc03661#Azure.Storage.StorageRequestFailedException

Decide if we're going to add an ErrorCode to the base RequestFailedException so we can optionally remove the StorageRequestFailedException.  We'll also move AdditionalInformation into Data as part of this, though we still want to customize the exception message to look like it does now.

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/8144

### CODE REVIEW

**Review by jsquire — APPROVED** [review]
The Event hubs bits look good to me.   I didn't notice anything in a glance at the wider set of changes, though I lack a deep understanding of the storage implementation.

I'd definitely advise having someone more familiar with storage approve as well.

File: sdk/storage/Azure.Storage.Common/src/StorageRequestFailedExceptionHelpers.cs [hunk 1]
```diff
@@ -14,69 +14,27 @@ namespace Azure.Storage
 {
     /// <summary>
     /// Provide helpful information about errors calling Azure Storage endpoints.
-    /// </summary>
-#pragma warning disable CA1032 // Implement standard exception constructors
-    public partial class StorageRequestFailedException : RequestFailedException
-#pragma warning restore CA1032 // Implement standard exception constructors
+    /// </summary>S
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Common/src/StorageRequestFailedExceptionHelpers.cs]: Nit - `S`

**Review by tg-msft — APPROVED** [review]
Looks good - thank you!

**Review by amnguye — APPROVED** [review]

**Comment by pakrym** [comment]
Checks are green in devops.

**Pull request merged**